py/picca/pk1d/postproc_pk1d.py

```diff
@@ -1212,21 +1221,21 @@ def compute_cov(
         p1d_groups_product_sum = np.nansum(
             [p1d_groups_product_sum, np.outer(p1d_group, p1d_group)], axis=0
         )
-        covariance_weights_product_of_sum = np.nansum(
+        covariance_weights_product_sum = np.nansum(
             [
-                covariance_weights_product_of_sum,
+                covariance_weights_product_sum,
                 np.outer(covariance_weights[i], covariance_weights[i]),
             ],
             axis=0,
         )
-        weights_product_of_sum = np.nansum(
-            [weights_product_of_sum, np.outer(p1d_weights[i], p1d_weights[i])], axis=0
+        weights_product_sum = np.nansum(
+            [weights_product_sum, np.outer(p1d_weights[i], p1d_weights[i])], axis=0
         )
 
     del p1d_groups, covariance_weights, p1d_weights
 
-    covariance_matrix = (weights_product_of_sum / weights_sum_of_product) * (
-        (p1d_groups_product_sum / covariance_weights_product_of_sum) - mean_pk_product
+    covariance_matrix = ((weights_sum_product /weights_product_sum) - 1)**(-1) * (
```

Comment:
the -1 in the factor here seems to be the second actual change (i.e. the correction for finite sample size aka 1/(N-1) factors in variance calculations)

Reply:
Overall I think this code could have some simplification/clarification. Maybe we should add the covariance document to picca and explicitely mention which quantities are which with equation numbers etc as comments.

Reply:
Yes this is the second change, which gives a unbiased version of the covariance. Same here no big impact. In the end the implementation and estimator will be detailled in the Y1 validation paper

Reply:
the issue for me is not where exactly the documentation is, but at least on first glance it's somewhat hard to follow the code even looking at JMs note next to it. So we should probably try to streamline/comment it more in the relevant parts, to help the next person (or your future self) getting here. E.g. sum_product and product_sum are somewhat easily confused (but I don't have a great suggestion for other naming conventions which is why I suggested comments).
That said, the change from the biased to unbiased estimator is finished in here, so lets merge whenever you're fine to do so